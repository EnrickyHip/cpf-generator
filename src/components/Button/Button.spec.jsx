import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Button } from ".";

describe("<Button />", () => {
  it("should render the button with the text 'Test'", () => {
    const fn = jest.fn();
    render(<Button onClick={fn}>Test</Button>);
    const button = screen.getByRole("button", { name: /test/i });

    expect(button).toBeInTheDocument();
  });

  /* -------------------------------- */

  it("should call function onClick", () => {
    const fn = jest.fn();
    render(<Button onClick={fn}>Test</Button>);

    const button = screen.getByRole("button", { name: /test/i });
    userEvent.click(button);

    expect(fn).toBeCalled();
  });

  /* -------------------------------- */

  it("should match snapshot", () => {
    const fn = jest.fn();
    const { container } = render(<Button onClick={fn}>Test</Button>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
