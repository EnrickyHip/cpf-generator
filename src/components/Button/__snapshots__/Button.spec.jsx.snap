// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<button
  class="btn"
  type="button"
>
  Test
</button>
`;
